From c2bd0a7e6caba9597e190fe4a8cfe0c8c128e72e Mon Sep 17 00:00:00 2001
From: Masatake YAMATO <yamato@redhat.com>
Date: Sun, 5 Jan 2020 13:54:50 +0900
Subject: [PATCH] lregex: fix meaningless type mismatching between unsigned int
 and unsigned long

Partially close #2369.

A pointer to unsigned int variable is passed to functions expecting
a pointer to unsigned long variable.

This didn't cause a trouble because the sizes of the types are the
same on x86_64, the most popular platform. However, on s390x, they
are different, and cause a trouble as reported in #2369.

Signed-off-by: Masatake YAMATO <yamato@redhat.com>
---
 main/lregex.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/main/lregex.c b/main/lregex.c
index 5e0745dd..243f4607 100644
--- a/main/lregex.c
+++ b/main/lregex.c
@@ -478,7 +478,7 @@ static flagDefinition prePtrnFlagDef[] = {
 static void scope_ptrn_flag_eval (const char* const f  CTAGS_ATTR_UNUSED,
 				  const char* const v, void* data)
 {
-	unsigned long *bfields = data;
+	unsigned int *bfields = data;
 
 	if (strcmp (v, "ref") == 0)
 		*bfields |= SCOPE_REF;
@@ -497,7 +497,7 @@ static void scope_ptrn_flag_eval (const char* const f  CTAGS_ATTR_UNUSED,
 static void placeholder_ptrn_flag_eval (const char* const f  CTAGS_ATTR_UNUSED,
 				     const char* const v  CTAGS_ATTR_UNUSED, void* data)
 {
-	unsigned long *bfields = data;
+	unsigned int *bfields = data;
 	*bfields |= SCOPE_PLACEHOLDER;
 }
 
